perfkitbenchmarker/configs/benchmark_config_spec.py

```diff
@@ -204,16 +205,16 @@ def _GetOptionDecoderConstructions(cls):
             'default': spark_service.PROVIDER_MANAGED,
             'valid_values': [spark_service.PROVIDER_MANAGED,
                              spark_service.PKB_MANAGED]}),
-        'num_workers': (option_decoders.IntDecoder, {
-            'default': 3, 'min': 1}),
+        'worker_group': (_VmGroupSpecDecoder, {}),
+        'master_group': (_VmGroupSpecDecoder,
+                         {'default': None,
+                          'none_ok': True}),
         'cloud': (option_decoders.EnumDecoder, {
```

Comment:
cloud is a property of the vmgroup spec, so pulled it from there. 
zone is not in the vmgroup spec, but may be in the vm spec, so pulled it from the flag or the master_group vmspec.
I didn't see where either contained project, so I just pulled that from the flag.